pkg/windows/windows.go

```diff
@@ -735,13 +734,19 @@ func (vm *windows) removeDirectories() error {
 }
 
 // transferFiles copies various files required for configuring the Windows node, to the VM.
-func (vm *windows) transferFiles() error {
+func (vm *windows) transferFiles(tlsData map[string][]byte) error {
 	vm.log.Info("transferring files")
 	for src, dest := range vm.filesToTransfer {
 		if err := vm.EnsureFile(src, dest); err != nil {
 			return fmt.Errorf("error copying %s to %s: %w", src.Path, dest, err)
 		}
 	}
+	if err := vm.EnsureFileContent(tlsData["tls.crt"], "tls.crt", TLSDir); err != nil {
+		return fmt.Errorf("unable to transfer TLS certs: %w", err)
+	}
+	if err := vm.EnsureFileContent(tlsData["tls.key"], "tls.key", TLSDir); err != nil {
+		return fmt.Errorf("unable to transfer TLS key: %w", err)
+	}
```

Comment:
consider pulling this out into a helper (and maybe making tls.crt and tls.key local constants). will allow you to remove duplicated calls in secret_controller.reconcileTLSSecret()

Reply:
Unsure if we should add it to the interface since it will not be a genric helper.